### DISCUSSION TITLE
Contour lines are completely offset

### DISCUSSION THREAD

**Discussion by DrYOSE** [arkypita/LaserGRBL#2149] [Q&A] [answered]
I am a complete newbie with LaserGRBL and apologize if my question has already been answered elsewhere.
I did not find anything that could help so far.

I have purchased a laser engraver lately that gave mixed results, partly because the hardware was of not so good quality, but probably also because the software settings are wrong. The hardware is supposed to be LaserGRBL compatible as it is advertised as such, comes with a version of LaserGRBL (though the latest version has been downloaded from lasergrbl.com and run under Windows). 

The biggest problem that I currently encounter is an obvious offset of the contour lines with regards to the filling (dark area). They should perfectly be superimposed, but they do not, as a matter of fact (see photo).
![image_offset](https://github.com/arkypita/LaserGRBL/assets/150358915/c821496c-4a16-4843-ab37-7a29d0d78a52)

Any guidance would be greatly appreciated.
Best regards,

NOTE: passthrough and line by line modes do not show this problem, obviously.

**Comment by Tribo-2dot0** [comment]
Você usou a mesma imagem para ambas os processos? Utilizou a ferramenta Auto Trim Image ou Crop Image em apenas uma das vezes que a carregou? Utilizou a função Append File para executar os dois processos de uma única vez ou o fez separadamente? Verificou seo os valores de Off-Set estavam iguais em ambos?

**Reply by DrYOSE** [reply]
That was ONE engraving only. It was NOT two separate processes. This process should have engraved the whole thing in one go, but the contour lines ended up being completely offset.

**Comment by arkypita** [comment, marked as answer]
If you have this effect is because your laser head is not returned at the original position after filling job.
This could be because it has some slippage in belt, some cumulated error, jam hitting some part or similar issue.

**Reply by DrYOSE** [reply]
Understood. The origin of the problem is the hardware. I should have purchased a better quality engraver.

Over long distances, the belt slippage causes an obvious offset, while over short distances (for instance line by line) that does not occur.
